lib/team.rb

```diff
@@ -0,0 +1,32 @@
+require 'pry'
+
+class Team
+  attr_reader :country,
+              :players,
+              :eliminated
+  def initialize(country, eliminated = false)
+    @country = country
+    @players = []
+    @eliminated = eliminated
+  end
+
+  def eliminated
+    @eliminated = true
+  end
+
+  def eliminated?
+   @eliminated 
+  end
+
+  def add_player(player)
+  @players << player
+  end
+
+  def player_by_position(position)
+    @players.find_all do |find|
+            
+      
+    #I know this isn't the right way...
```

Comment:
It is! Very close! Just need to finish it off!

```ruby
@players.find_all do |player|
  player.position == position
end
```

One thing to remember: the word that you put between the pipes (for example `|player|` in my code above) represents a *single element from the array.* So if I have an array holding a collection of players, I would probably use `player` in between those pipes to represent a single player. Let me know if that makes sense!